harbour-shmoose.pro

```diff
@@ -7,8 +7,8 @@ contains(DEFINES, SFOS) {
 }
 
 # from swift-config
-SWIFTCXX = -DSWIFTEN_STATIC -DBOOST_ALL_NO_LIB -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_SIGNALS_NO_DEPRECATION_WARNING -DSWIFT_EXPERIMENTAL_FT
-SWIFTLIB = -lSwiften -lSwiften_Boost -lrt -lz -lssl -lcrypto -lxml2 -lresolv -lpthread -ldl -lm -lc -lstdc++
+SWIFTCXX = $$system("swiften-config --cflags")
+SWIFTLIB = $$system("swiften-config --libs")
```

Comment:
In my opinion it's better to use a direct relativ call. Most of the people didn't have swiften globally installed. So, something like this will work on any system which has swift compiled:

SWIFTCXX = $$system("$${SWIFT3PATH}/Swiften/Config/swiften-config--cflags")

Reply:
But your way of doing that will break the build for those who have, for example me building on the my Arch Linux. 

Those people who have swiften built locally can just do 
```
export PATH=${SWIFT3PATH}/Swiften/Config/
qmake
make
```

Reply:
As far as I know Arch Linux is the only distribution who did such a consequent handling of self compiled software as packages. All the other 'traditional' ways of handling self compiled source code with configure, make, make install must do additional steps or configurations.

Maybe there is functionality within the qt pro file to check for the swiften-config binary in PATH and if not, use the relativ file?

Reply:
For deb/rpm distros there is checkinstall, which allows you to create a deb package out of make install.
https://help.ubuntu.com/community/CheckInstall

and in most cases it's not a problem, since the distro already packages swiften as a precompiled binary: https://packages.ubuntu.com/search?keywords=swiften&searchon=names&suite=artful&section=all

Re: subdirectory. And what is the standard path for the local swift library? `../swift` or `../swift-3.0` or `../swift-3.0-host` or just a subdirectory in the three variants above? There's a lot of freedom on the side of the compiling party.

I'd see this a better solution: if the environmental variable `SWIFT3PATH` (or even `SWIFTPATH` in the future, see #33) is set, we use a local swift build with a local swiften config. Otherwise, we use a global one. It's mostly like overriding the C compiler with the `CC` env variable.

Reply:
In Debian stable, the provided swiften package is too old. And even worse it's not PIC compiled. One had to patch the scons file for SFOS target to get it to work.

My idea about the relativ link to swiften-config didn't work. The mounted share from host system is mounted not executable.

If you do a description in the Readme on how to package (arm and x86) the patched swiften source for debian stable, I will accept this PR.

Reply:
There are repositories provided for Ubuntu and Debian by swiften itself: https://swift.im/releases.html
Is that enough?

Reply:
One more note - it may be a good idea to wait with accepting this PR until I get swiften packaged for SFOS. That, combined with thia PR would greatly reduce the entry effort.

Reply:
The available swiften 3 package for x86 is a good starting point. Now only the patched arm package is missing.